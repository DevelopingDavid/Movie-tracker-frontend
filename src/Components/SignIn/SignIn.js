import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';

export class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      error: ''
    }
  }

  fetchUsers = async () => {
    const userInfo = this.state;
    const options = {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(userInfo)
    }
    const userDatabase = 'http://localhost:3000/api/users';
    const response = await fetch(userDatabase, options);
    if(response.ok) {
      const user = await response.json();
      return user.data
    } else {
      return 'Incorrect email/password combination'
    }
  }

  validateUser = async (e) => {
    e.preventDefault()
    const user = await this.fetchUsers();
      if(typeof user === 'object') {
        console.log(this.props.loginUser)
        // this.props.loginUser(true)
      } else {
        this.setState({ error: user });
      }
  }

  saveInput = (event) => {
    event.preventDefault();
    if (event.target.name === 'password') {
      this.setState({
        password: event.target.value
      });
    } else if (event.target.name === 'email') {
      this.setState({
        email: event.target.value
      })
    
    } else if(event.target.name === 'name') {
      this.setState({
        name: event.target.value
      })
    } ;
  }

  render() {
    return (
      <div>
        Sign-In Form
        <form className='sign-in' onChange={this.saveInput} onSubmit={this.validateUser}>
          <input type='email' name='email' value={this.state.email} placeholder='email@example.com'/>
          <input type='password' name='password' value={this.state.password} placeholder='password'/>
          <input type='submit'/>
        </form>
        <NavLink to='/sign-up'>Sign Up</NavLink>
        { this.state.error && this.state.error } 
      </div>
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  loginUser: (loggedIn) => dispatch(loginUser(loggedIn))
});

export default connect(null, mapDispatchToProps)(SignIn);