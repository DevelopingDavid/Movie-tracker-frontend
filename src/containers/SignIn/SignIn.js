import React, { Component } from 'react';
import { NavLink, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import { loginUser } from '../../actions';

export class SignIn extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: '',
      error: ''
    }
  }

  fetchUsers = async () => {
    const userInfo = this.state;
    const options = {
      method: "post",
      headers: {
        "Content-Type": "application/json"
      },
      body: JSON.stringify(userInfo)
    }
    const userDatabase = 'http://localhost:3000/api/users';
    const response = await fetch(userDatabase, options);
    if(response.ok) {
      const user = await response.json();
      return user.data
    } else {
      return 'Incorrect email/password combination'
    }
  }

  validateUser = async (e) => {
    e.preventDefault()
    const user = await this.fetchUsers();
    if(typeof user === 'object') {
      this.props.loginUser(user.id, user.name);
      localStorage.setItem('movieTrackerUser', JSON.stringify(this.props.user))
    } else {
      this.setState({ error: user });
    }
  }

  saveInput = (event) => {
    event.preventDefault();
    const {name, value} = event.target;
    this.setState({
      [name]: value
    });
  }

  render() {
    if(Object.keys(this.props.user).length > 0) { 
      return <Redirect to='/movies'/>
    }
    
    return (
      <div className='sign-in-container'>
        <section>
          <form className='sign-in-form' onSubmit={this.validateUser}>
            <div>
              <label>Email</label>
              <i className="fas fa-envelope"></i>
            </div>
            <input type='email' name='email' value={this.state.email} placeholder='email@example.com' onChange={this.saveInput} autoComplete="off"/>
            <div>
              <label>Password</label>
              <i className="fas fa-lock"></i>
            </div>
            <input type='password' name='password' value={this.state.password} placeholder='password' onChange={this.saveInput} autoComplete="off"/>
            <input type='submit'/>
          </form>
          <NavLink className='sign-up-btn' to='/sign-up'>Sign Up</NavLink>
          <div>
            { this.state.error && this.state.error } 
          </div>
        </section>
      </div>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  loginUser: (id, name) => dispatch(loginUser(id, name))
});

export const mapStateToProps = (state) => ({
  user: state.user
});

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);