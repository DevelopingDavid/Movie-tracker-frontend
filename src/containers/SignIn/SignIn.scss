.sign-in-container {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;
  section {
    height: 50%;
    background-color: #0e0f11;
    width: 50%;
    color: white;
    border-radius: 1.7rem;
    font-family: sans-serif;
    font-size: 2rem;
    border: solid 3px #940002;
  }
  form {
    margin: 100px 100px 0 100px;
  }
  input {
    margin: 20px 0;
    border: none;
    border-bottom: solid 2px white;
    background: none;
    font-size: 2rem;
    width: 100%;
    outline: none;
    padding-bottom: 5px;
    color: white;
  }
  .sign-in-btn {
    background: white;
    color: black;
  }
  label {
    margin-right: 14px;
  }
}

.sign-in-form {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.sign-up-link {
  color: white;
}