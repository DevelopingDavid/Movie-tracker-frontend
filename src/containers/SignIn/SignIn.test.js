import React from 'react';
import SignIn from '../SignIn/SignIn';
import { shallow } from 'enzyme';

describe('SignIn', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow( 
      <SignIn/>
    ) 
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});