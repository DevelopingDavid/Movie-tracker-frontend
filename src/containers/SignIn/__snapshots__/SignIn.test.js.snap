// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn
    loginUser={[MockFunction]}
    user={
      Object {
        "email": "david@gmail.com",
        "id": 1,
        "name": "david",
        "password": "password",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "to": "/movies",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "to": "/movies",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
